jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java

```diff
@@ -31,167 +30,115 @@
-
+    // Then
     Deployment deployment = (Deployment) builder.buildFirstItem();
-    Map<String, String> selectors = deployment.getSpec().getSelector().getMatchLabels();
-    assertThat(selectors)
-            .doesNotContainKey("version")
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","my-custom-app-name");
+    assertThat(deployment)
+        .extracting("spec.selector.matchLabels")
+        .asInstanceOf(InstanceOfAssertFactories.MAP)
+        .containsEntry("group", "groupId")
+        .containsEntry("app", "my-custom-app-name")
+        .doesNotContainKey("version")
+        .doesNotContainKey("project");
   }
 
   @Test
-  public void testEmptyCustomAppName() {
-    // Setup
+  void create_emptyCustomAppName() {
+    // Given
     properties.setProperty("jkube.enricher.jkube-project-label.app", "");
-
-    KubernetesListBuilder builder = createListWithDeploymentConfig();
-    projectLabelEnricher.enrich(PlatformMode.kubernetes, builder);
-    KubernetesList list = builder.build();
-
-    Map<String, String> labels = list.getItems().get(0).getMetadata().getLabels();
-    assertThat(labels)
-            .isNotNull()
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","")
-            .containsEntry("version","version");
-    builder = new KubernetesListBuilder().withItems(new DeploymentBuilder().build());
+    KubernetesListBuilder builder = new KubernetesListBuilder().withItems(new DeploymentBuilder().build());
+    // When
     projectLabelEnricher.create(PlatformMode.kubernetes, builder);
-
+    // Then
     Deployment deployment = (Deployment) builder.buildFirstItem();
-    Map<String, String> selectors = deployment.getSpec().getSelector().getMatchLabels();
-    assertThat(selectors)
-            .doesNotContainKey("version")
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","");
+    assertThat(deployment)
+        .extracting("spec.selector.matchLabels")
+        .asInstanceOf(InstanceOfAssertFactories.MAP)
+        .containsEntry("group", "groupId")
+        .containsEntry("app", "")
+        .doesNotContainKey("version")
+        .doesNotContainKey("project");
   }
 
   @Test
-  public void testDefaultAppName() {
-    KubernetesListBuilder builder = createListWithDeploymentConfig();
-    projectLabelEnricher.enrich(PlatformMode.kubernetes, builder);
-    KubernetesList list = builder.build();
-
-    Map<String, String> labels = list.getItems().get(0).getMetadata().getLabels();
-
-    assertThat(labels).isNotNull()
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","artifactId").containsEntry("version","version");
-
-    builder = new KubernetesListBuilder().withItems(new DeploymentBuilder().build());
+  void create_defaultAppName() {
```

Comment:
You need to create an additional test for DeploymentConfig, the test you refactored was testing both Deployment and DeploymentConfig